right-sidebar: Show orange/idle dot for personal presence.

Harshbansal8705 opened the pull request:
This commit ensures that users see their own presence status as orange/idle when applicable matching what other users see. Previously, the system only updated offline status but did not reflect idle status correctly. This change improves consistency in presence indications reducing confusions.

Earlier, the user was seeing himself as active if he's sharing the presence. This commit adds one more condition to check if the user is idle or not. If the user is idle, it'll show idle, otherwise it'll show active.

Fixes: #18846

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
![Screenshot_20250201_035945](https://github.com/user-attachments/assets/980dfc32-208e-421a-b4b7-f9e150d4d8cd)


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

Harshbansal8705 committed:
right-sidebar: Show orange/idle dot for personal presence.

This commit ensures that users see their own presence status
as orange/idle when applicable matching what other users see.
Previously, the system only updated offline status but did
not reflect idle status correctly. This change improves
consistency in presence indications reducing confusions.

Earlier, the user was seeing himself as active if he's sharing
the presence. This commit adds one more condition to check if
the user is idle or not. If the user is idle, it'll show idle,
otherwise it'll show active.

Fixes: #18846

alya commented:
@timabbott Do you want to review this small change?

timabbott commented:
As detailed in the original issue, we wanted to test the UX on chat.zulip.org before deciding if we indeed want to make this change, so tagging this for that.

timabbott commented:
Test-deploying on chat.zulip.org.

Harshbansal8705 commented:
Sorry, pressed the close button by mistake!

Harshbansal8705 commented:
As per the [CZO discussion](https://chat.zulip.org/#narrow/channel/6-frontend/topic/orange.20dot.20for.20self.20profile/with/2109668), there's only one thing remaining, which I'm trying to resolve, about how to update the self-user status instantly.

Harshbansal8705 committed:
presence: Update idle-to-active status immediately for self-user.

Previously, when the user transitioned from idle to active, the
presence status was updated after the `presence` request was sent at
the intervals of one minute, which could cause delay up to 1 minute.

This commit updates the user status as soon as presence request is
sent and response is recieved from the server using redraw_user.

Note: activity.initialize() function had to be moved to
activity_ui.initialize_activity() to avoid cyclic-import error.

Co-authored-by: reharsh

Harshbansal8705 commented:
@timabbott I think it's ready for review now. I've completed the test cases coverage with the help of @reharsh. Please take a look.

Harshbansal8705 commented:
@timabbott Just wanted to remind you about this PR.

karlstolley commented on web/src/activity.ts:
```diff
@@ -5,6 +5,7 @@ import {z} from "zod";
 import * as channel from "./channel.ts";
 import {electron_bridge} from "./electron_bridge.ts";
 import {page_params} from "./page_params.ts";
+import {maybe_get_user_by_id, my_current_user_id} from "./people.ts";
```
I think we're trying to get away from import statements like this and instead using `*`, so that it's easier to grep for `people.maybe_get_user_by_id`

Harshbansal8705 replied:
Got it! I've updated it.
